File: easy/go/0028.go
func strStr(haystack string, needle string) int {
    if len(needle) == 0{
        return 0
    }
    var sz int = len(haystack)
    var t_sz int = len(needle)
    for i := 0; i < sz - t_sz + 1; i++{
        var find bool = true
        for j := 0; j < t_sz; j++{
            if haystack[i+j] != needle[j]{
                find = false
                break
            }
        }
        if find{
            return i
        }
    }
    return -1
}

File: easy/c/0028.c
int strStr(char * haystack, char * needle){
    if(strlen(needle) == 0){
        return 0;
    }
    int target_size = strlen(needle);
    int size = strlen(haystack);
    for(int i = 0 ; i < size - target_size + 1; i++){
        bool find = true;
        for(int j = 0 ; j < target_size; j++){
            if(haystack[i+j] != needle[j]){
                find = false;
                break;
            }
        }
        if(find){
            return i;
        }
    }
    return -1;
    
}